packages/workbox-precaching/controllers/PrecacheController.mjs

```diff
@@ -176,4 +177,90 @@ export default class PrecacheController {
 
     return true;
   }
+
+  /**
+   * Compare the URL's and determines which assets are no longer required
+   * in the cache.
+   *
+   * This should be called in the service worker activate event.
+   *
+   * @return {Promise<object>} Resolves with an object containing details
+   * of the deleted cache requests and precache revision details.
+   */
+  async cleanup() {
+    const expectedCacheUrls = [];
+    this._entriesToCacheMap.forEach((entry) => {
```

Comment:
A map would be more concise here I think:

```js
const expectedCacheUrls = this._entriesToCacheMap((entry) => entry._cacheRequest.url);
```

Reply:
Either I'm missing something or your code is missing `.map` because I just tried using a Map object like this and got an error.

`_entriesToCacheMap` is a Map object, so it has no `.map` method if that is what you wanted to change.

Reply:
Ahh, ok, I was thinking it was an array. Carry on!